main_loop_tf/validate.py

```diff
@@ -280,6 +299,8 @@ def write_summary(labs, val):
 def save_images(img_queue, save_basedir, sentinel):
     import matplotlib as mpl
     import seaborn as sns
+    from utils import flowToColor
+    import cv2
```

Comment:
I don't like cv2 to be a mandatory requirement for main_loop_tf. Move the import where you use it please, and try not to whenever is possible :)